### PR TITLE
Attribute-authority: update decryption key naming convention

### PR DESCRIPTION
**Pull request by dulalsaurab**

### CODE REVIEW

File: src/consumer.cpp [hunk 1]
```diff
@@ -93,7 +96,7 @@ Consumer::consume(const Name& dataName,
                   const ErrorCallback& errorCallback)
 {
   Interest interest(dataName);
-  interest.setMustBeFresh(true);
+  // interest.setMustBeFresh(true);
```
Comment (tianyuan129) [inline src/consumer.cpp]: Could you write comments inline to explain why this flag should not be set?

File: src/consumer.cpp [hunk 2]
```diff
@@ -143,7 +146,8 @@ Consumer::decryptContent(const Data& data,
   Name ckName(encryptedContent.get(tlv::Name));
   NDN_LOG_INFO("CK Name is " << ckName);
   Interest ckInterest(ckName);
-  ckInterest.setMustBeFresh(true);
+
```
Comment (tianyuan129) [inline src/consumer.cpp]: Remove the empty line.

File: src/producer.cpp [hunk 3]
```diff
@@ -109,7 +109,7 @@ Producer::ckDataGen(const Policy& accessPolicy, std::shared_ptr<Data> dataTempla
     auto ckData = std::move(dataTemplate);
     ckData->setName(ckDataName);
     ckData->setContent(contentKey->makeCKContent());
-    m_keyChain.sign(*ckData, signingByCertificate(m_cert));
+    m_keyChain.sign(*ckData, signingWithSha256());
```
Comment (tianyuan129) [inline src/producer.cpp]: This doesn't help anything. If we want to provide flexibility in signing, we should provide a signingInfo as an input parameter.
See [ndn-svs](https://github.com/named-data/ndn-svs/blob/master/examples/chat-pubsub.cpp#L45).

File: src/producer.cpp [hunk 4]
```diff
@@ -165,7 +165,7 @@ Producer::ckDataGen(const std::vector<std::string>& attributes, std::shared_ptr<
...
```
Comment (tianyuan129) [inline src/producer.cpp]: Same.

**Review by tianyuan129 — COMMENTED** [review]
Some fixes need to be done. See individual comments.

**Comment by tylerliu** [comment]
CI does not pass.

**Comment by tianyuan129** [comment]
> CI does not pass.

cuz unit tests haven't been updated according to the new naming convention.

File: src/consumer.cpp (hunk 1 above)
Comment (dulalsaurab) [reply src/consumer.cpp]: I can, but first, we need justification on why it was/ or should be MustBeFresh. Most of the applications don't want to sever just the fresh data, so what's the reason for having it mustbefresh in the first place.

Comment (tianyuan129) [reply src/consumer.cpp]: I can see two approaches:
1/ Easy fix: Removing this line and leaving a comment to explain the tradeoff
2/ Longer term: letting user to specify Interest parameters.

Either is fine to me.

**Review by tianyuan129 — COMMENTED** [review]
Fine to me, but the CI failed still. Have you updated the examples accordingly?
Also please add the description for this PR. It's still empty right now.

We are close to wrap this big patch up 🙂

**Comment by dulalsaurab** [comment]
I am testing the path with mGuard, need to wait until my testing is done to proceed further.

**Comment by dulalsaurab** [comment]
> I am testing the path with mGuard, need to wait until my testing is done to proceed further.

Integration with mGuard passed. Up to you guys.

**Comment by tianyuan129** [comment]
@tylerliu seems that we're having problems on macOS. I believe this PR doesn't touch any platform specific things. How we should proceed? Merging and solve the CI issue later?

**Comment by tylerliu** [comment]
#25 working on CI changes here

**Comment by tianyuan129** [comment]
Hmmm are you suggesting we should proceed to merge? Or waiting #25 finishes first (which is another long story).

**Review by tianyuan129 — APPROVED** [review]
Okay to merge, need some final confirmations.

**Pull request merged**